lib/path.js

```diff
@@ -185,14 +185,13 @@ win32.isAbsolute = function(path) {
 };
 
 win32.join = function() {
-  function f(p) {
-    if (typeof p !== 'string') {
-      throw new TypeError('Arguments to path.join must be strings');
-    }
-    return p;
+  var paths = [];
```

Comment:
Perhaps specifying the length up front, `var paths = new Array(arguments.length);` and assigning at the index would be faster?